File: PRACTICA3/QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author jesus
 */
public class Casilla {
    private int numeroCasilla;
    private int coste;
    private TipoCasilla tipo;
    private TituloPropiedad titulo;

    public Casilla(int numeroCasilla,TipoCasilla tipo) {
        this.numeroCasilla = numeroCasilla;
        this.coste = 0;
        this.tipo = tipo;
        this.titulo = null;
    }

    //Constructor para Casillas del TipoCasilla -> CALLE
    public Casilla(int numeroCasilla,TituloPropiedad titulo) {
        this.numeroCasilla = numeroCasilla;
        setTitulo(titulo);
        this.tipo = TipoCasilla.CALLE;
        this.coste = this.titulo.getPrecioCompra();
    }
    
    TituloPropiedad asignarPropietario(Jugador jugador)
    {
        titulo.setPropietario(jugador);
        
        return titulo;
    }
    
    int pagarAlquiler()
    {
        int costeAlquiler = titulo.pagarAlquiler();
        return costeAlquiler;
    }
    
    boolean PropietarioEncarcelado()
    {
        return titulo.propietarioEncarcelado();
    }
    
    boolean soyEdificable()
    {
        return (tipo == TipoCasilla.CALLE);
    }
    
    boolean tengoPropietario()
    {
        return titulo.tengoPropietario();
    }

    int getNumeroCasilla() {
        return numeroCasilla;
    }

    int getCoste() {
        return coste;
    }

    TipoCasilla getTipo() {
        return tipo;
    }

    TituloPropiedad getTitulo() {
        return titulo;
    }

    private void setTitulo(TituloPropiedad titulo) {
        this.titulo = titulo;
    }

    @Override
    public String toString() {
        return "Casilla{" + "numeroCasilla=" + numeroCasilla + ", coste=" + coste + ", tipo=" + tipo + ", titulo=" + titulo + '}' + "\n";
    }
    
    
    
    
    
}

File: PRACTICA3/QytetetRuby/lib/casilla.rb
#encoding: utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "tipo_casilla.rb"
require_relative "titulo_propiedad.rb"

module ModeloQytetet
  class Casilla
    
    attr_reader :numero_casilla,:tipo,:titulo,:coste
    attr_writer :titulo
    
    def initialize(numero_casilla,tipo,titulo)
      @numero_casilla = numero_casilla
      if tipo == TipoCasilla::CALLE
        @tipo = TipoCasilla::CALLE
        @titulo = titulo
        @coste = titulo.precio_compra
      else
        @tipo = tipo
        @titulo = 0
        @coste = 0
        
      end
    end
    
    def asignar_propietario(jugador)
    
        @titulo.propietario = jugador
        
        return @titulo
    end
    
    def pagar_alquiler
    
        coste_alquiler = @titulo.pagar_alquiler
        return coste_alquiler
    end
    
    def propietario_encarcelado
    
        return titulo.propietario_encarcelado
    end
    
    def soy_edificable
    
        return (@tipo == TipoCasilla::CALLE)
    end
    
    def tengo_propietario
    
        return @titulo.tengo_propietario
    end

    
    def to_s
      "N_casilla: #{@numero_casilla} \n Tipo: #{@tipo} \n Titulo: #{@titulo} \n Coste: #{@coste} \n\n"
    end
  end
end